packages/runtime/src/widgets/Stepper/Stepper.tsx

```diff
@@ -51,35 +53,35 @@ interface CustomConnectorProps {
 }
 
 const Stepper: React.FC<StepperProps> = (props) => {
-  const [activeStep, setActiveStep] = useState(props.activeStepIndex ?? 0);
+  const [activeStep, setActiveStep] = useState(props?.activeStepIndex);
+
   const itemTemplate = props["item-template"];
   const { namedData } = useTemplateData({ ...itemTemplate });
   const stepsLength = namedData.length;
   const stepTypes = itemTemplate?.template;
   const handleNext = useCallback(() => {
-    if (activeStep < namedData.length - 1) {
-      setActiveStep((prevActiveStep) => prevActiveStep + 1);
+    if (activeStep && activeStep < namedData.length - 1) {
+      setActiveStep(activeStep + 1);
     }
   }, [activeStep, namedData.length]);
-
   const handleBack = useCallback(() => {
-    if (activeStep !== 0) {
-      setActiveStep((prevActiveStep) => prevActiveStep - 1);
+    if (activeStep && activeStep !== 0) {
+      setActiveStep(activeStep - 1);
     }
   }, [activeStep]);
   const handleStep = (step: number) => () => {
     setActiveStep(step);
   };
-  const { values, rootRef } = useRegisterBindings(
-    { ...props, activeStep },
-    props.id,
-    {
-      handleNext,
-      handleBack,
-    },
-  );
+  const { values, rootRef } = useRegisterBindings({ ...props }, props.id, {
+    handleNext,
+    handleBack,
+  });
+  useEffect(() => {
+    if (isString(activeStep) && isExpression(activeStep)) {
```

Comment:
You can check the props value so you don't need to overload your state variable. Your state variable should always just be an integer